src/main.cpp

```diff
@@ -28,20 +27,47 @@ int Main(vector<string> args)
 		%whitespace <- [ \t]*
 	)";
 
+	// create parser
 	parser parser(syntax);
 
 	if (!parser){
 		cout << "error creating grammar" << endl;
 		return EXIT_FAILURE;
 	}
 
-	parser.enable_packrat_parsing(); // Enable packrat parsing.
+	// create callbacks for debugging
+	auto rules = parser.get_rule_names();
 
-	int val;
+	for(auto r:rules){
+		parser[r.c_str()].enter = [r](const char* s, size_t n, any& dt) {
+			auto& indent = *dt.get<int*>();
+			for(int i=0; i<indent; i++){cout << "|  ";}
```

Comment:
Consider writing a function here and below like

    std::ostream& repeat(std::ostream& os, const char* s, int num);

Reply:
Thanks for your suggestion! Wouldst thou care to elaborate, pretending I didn't get it?